harness/o.n.insane/src/org/netbeans/insane/impl/InsaneEngine.java

```diff
@@ -51,23 +52,23 @@ public InsaneEngine(Filter f, Visitor v, boolean analyzeStatic) {
 
     public InsaneEngine(ObjectMap backend, Filter f, Visitor v, boolean analyzeStatic) {
         objects = backend;
-        filter = f == null ? ScannerUtils.noFilter() : f;
+        filter = (f != null ? f : ScannerUtils.noFilter());
         visitor = v;
         analyzeStaticData = analyzeStatic;
     }
 
         
     // normal set is enough, java.lang.Class have identity-based equals()
-    private Set<Class> knownClasses = new HashSet<Class>();
+    private final Set<Class<?>> knownClasses = new HashSet<>();
     
     // The queue for BFS scan of the heap.
     // each thing, before added to the queue, is added
     // to the known* structures and reported to the visitor
-    private Queue<Object> queue = new Queue<Object>();
+    private final Queue<Object> queue = new Queue<>();
     
-    public void traverse(Collection roots) throws Exception {
+    public void traverse(Collection<?> roots) throws Exception {
```

Comment:
If this is exported, please don't add generics for general cleanup.

Reply:
reverted